.github/actions/happy-cleanup/action.yml

```diff
@@ -29,12 +32,19 @@ runs:
         ENV: ${{ inputs.env }}        
         TFE_TOKEN: ${{ inputs.tfe_token }}
         TIME: ${{inputs.time}}
+        EXCLUDE: ${{ inputs.exclude }}
       shell: bash
       run: |
         set -ue
         set -o pipefail
 
         date=`date +%Y-%m-%d'T'%H:%M'Z' -d "$TIME ago"`
+        if [[ ! -z ${EXCLUDE} ]]; then
+          for i in $(happy list --aws-profile "" --output json --env $ENV | jq -r --arg date "$date" --arg exclude "$EXCLUDE" '.[] | select(.last_updated < $date) | select(any(.stack; contains($exclude))|not) | .stack'); do
+            happy delete $i --env $ENV --aws-profile ""
+          done
+          exit
+        fi
         for i in $(happy list --aws-profile "" --output json --env $ENV | jq -r --arg date "$date" '.[] | select(.last_updated < $date) | .stack'); do
```

Comment:
do you need to delete this for loop? otherwise I think it will still do the same behavior as before

Reply:
also I'm not a super good reviewer of bash scripts, it looks like it would work. Might be a good idea to make a test branch in the happy repo and setup a test with the examples folder.

Reply:
or maybe it would just be faster to merge and try it out 🤷🏻 . up to you!

Reply:
@jakeyheath I don't need to delete the second forloop because it should exit before the second loop executes.

I can set up a test branch to test this based of the sha

Reply:
In looking into setting up test branches for this I am wondering if it would be faster to merge and test. I cleaned up many of the superfluous stacks last week so I don't have a big pool to pull from but I can demonstrate the outputs of these commands in the test plan

Reply:
I was able to test the command and am confident it will work